packages/relay/src/lib/clients/mirrorNodeClient.ts

```diff
@@ -1265,12 +1272,67 @@ export class MirrorNodeClient {
+        )} retry count ${i} of ${repeatCount}. Waiting ${
+          MirrorNodeClient.MIRROR_NODE_RETRY_DELAY
+        } ms before repeating request`,
       );
 
       // Backoff before repeating request
-      await new Promise((r) => setTimeout(r, this.MIRROR_NODE_RETRY_DELAY));
+      await new Promise((r) => setTimeout(r, MirrorNodeClient.MIRROR_NODE_RETRY_DELAY));
     }
     return result;
   }
+
+  /**
+   * Retrieves and processes transaction record metrics from the mirror node based on the provided transaction ID.
+   *
+   * @param {string} transactionId - The ID of the transaction for which the record is being retrieved.
+   * @param {string} callerName - The name of the caller requesting the transaction record.
+   * @param {string} requestId - The unique identifier for the request, used for logging and tracking.
+   * @param {string} txConstructorName - The name of the transaction constructor associated with the transaction.
+   * @param {string} operatorAccountId - The account ID of the operator, used to calculate transaction fees.
+   * @returns {Promise<{transactionFee: number;} | undefined>} - An object containing the transaction fee if available, or `undefined` if the transaction record is not found.
+   */
+  public async getTransactionRecordMetrics(
+    transactionId: string,
+    callerName: string,
+    requestId: string,
+    txConstructorName: string,
+    operatorAccountId: string,
+  ): Promise<ITransactionRecordMetric | undefined> {
+    const formattedRequestId = formatRequestIdMessage(requestId);
+
+    this.logger.trace(
+      `${formattedRequestId} Get transaction record via mirror node: transactionId=${transactionId}, txConstructorName=${txConstructorName}, callerName=${callerName}`,
+    );
+
+    const mirrorNodeRetries = parseNumericEnvVar(
+      'MIRROR_NODE_GET_CONTRACT_RESULTS_RETRIES',
+      'MIRROR_NODE_GET_CONTRACT_RESULTS_DEFAULT_RETRIES',
+    );
+
+    // poll mirror node to get transaction record
+    const transactionRecord = await this.repeatedRequest(
+      this.getTransactionById.name,
+      [transactionId, 0],
+      mirrorNodeRetries,
+      formattedRequestId,
+    );
+
+    if (!transactionRecord) {
+      this.logger.warn(
+        `${formattedRequestId} No transaction record retrieved: transactionId=${transactionId}, txConstructorName=${txConstructorName}, callerName=${callerName}`,
+      );
+    } else {
+      const transactionReceipt: IMirrorNodeTransactionRecord = transactionRecord.transactions.find(
+        (tx: any) => tx.transaction_id === formatTransactionId(transactionId),
+      );
+
+      const mirrorNodeTxRecord = new MirrorNodeTransactionRecord(transactionReceipt);
+
+      // get transactionFee
+      const transactionFee = getTransferAmountSumForAccount(mirrorNodeTxRecord, operatorAccountId);
+      return { transactionFee, txRecordChargeAmount: 0, gasUsed: 0 };
```

Comment:
- Move the `getTransferAmountSumForAccount` method inside the `MirrorNodeTransaction` class
- Rename `txRecordChargeAmount` to `transactionRecordFee`
```typescript
      // get transactionFee
      const transactionFee = transaction.getTransferAmountSumForAccount(operatorAccountId);
      return { transactionFee, transactionRecordFee: 0, gasUsed: 0 };
```

Reply:
@victor-yanev keep in mind that `getTransferAmountSumForAccount()` is also used in getTransactionRecordMetrics() method in sdkClient. I think keeping the getTransferAmountSumForAccount() method in a mutual place is better than include it in mirror-node-specific-class.

Reply:
Yes, but right now this utility method contains two different implementations inside of it:
```typescript
const getTransferAmountSumForAccount = (
  transactionRecord: TransactionRecord | MirrorNodeTransactionRecord,
  accountId: string,
): number => {
  if (transactionRecord instanceof MirrorNodeTransactionRecord) {
    // implementation for MirrorNodeTransactionRecord
  } else {
    // implementation for SDK client's TransactionRecord
  }
};
```

We can:
- extract the relevant part for `MirrorNodeTransactionRecord` inside `MirrorNodeClient` (or directly inside `MirrorNodeTransactionRecord`)
- and extract the relevant part for the SDK `TransactionRecord` inside `SdkClient`

We won't need to do any type checking, the two methods will implement it's own logic, and only the relevant part of the logic will be in the class where it belongs.

Reply:
Ah yeah it makes sense now! Yes I like that idea! Separating them into their own classes is a clean solution!